app/controllers/packages_controller.rb

```diff
@@ -18,6 +18,26 @@ def index
            meta: { totalResults: @packages.totalResults }
   end
 
+  def create
+    package_validation = Validation::CustomPackageParameters.new(package_params)
+
+    if package_validation.valid?
+      package_response = packages.create rm_api_package
+
+      # Instead of trying to merge objects together, lets just request
+      # the newly created package from RM API and render that.
```

Comment:
A good strategy, but should be handled by the model layer. In other words, the packages api should do the re-request if necessary.

Our move is gradually towards decoupling `mod-kb-ebsco` resources from flexirest resources (as in the case of `CustomLabel`) so this might be an option here.

Reply:
So by model layer here, I don't mean the RMAPI resource, I mean the controller's model. Right now, they are the same thing, and we are feeling a lot of pain because of it.

We want an abstract model layer whose primary allegiance is to capturing our app's REST API. In that world, the "create" method of the model, would decompose into _two_ requests of the RMAPI resource. If we can decouple our model in this way, then we can later ablate Flexirest without having to change the controller or serialization layers at all.
For example, we could move the current `Package` class into `RMAPIPackage`, and then make the `Package` model look like:

```ruby
class Package
  def self.create(attributes)
    package_response = rmapi_packages.create attributes
    rmapi_package = rmapi_packages.find(rmapi_package_id_for package_response)
    return new_from_rmapi_package rm_api_package
  end
end
```

In general, our life is going to be a lot easier if we don't account for RMAPI in our controllers at all. I don't know if this is possible, but I would time box it for a couple hours, and then proceed if it isn't feasible after that investment.